ActivatePubKey/__tests__/handler.test.ts

```diff
@@ -0,0 +1,525 @@
+    );
+
+    upsertMock.mockImplementationOnce(() => {
+      return TE.right(aValidRetrievedPopDocument);
+    });
+
+    upsertMock.mockImplementationOnce(() => {
+      return TE.right(aValidRetrievedPopDocument);
+    });
+
+    const activatePubKeyHandler = ActivatePubKeyHandler(
+      getPopDocumentReader(lollipopPubKeysModelMock),
+      getPopDocumentWriter(lollipopPubKeysModelMock),
+      getAssertionWriter(
+        blobServiceMock,
+        LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME
+      )
+    );
+
+    const aValidActivatePubKeyPayload: ActivatePubKeyPayload = {
+      fiscal_code: aFiscalCode,
+      expires_at: new Date(),
+      assertion_type: AssertionTypeEnum.SAML,
+      assertion: "" as NonEmptyString
+    };
+
+    const assertionRefsResult = await getAllAssertionsRef(
+      JwkPubKeyHashAlgorithmEnum.sha512,
+      JwkPubKeyHashAlgorithmEnum.sha256,
+      aValidJwk
+    )();
+
+    if (assertionRefsResult._tag === "Left") fail();
+
+    const res = await activatePubKeyHandler(
+      contextMock,
+      aValidSha256AssertionRef,
+      aValidActivatePubKeyPayload
+    );
+
+    expect(findLastVersionByModelIdMock).toHaveBeenCalledTimes(1);
+    expect(findLastVersionByModelIdMock).toHaveBeenCalledWith([
+      aValidSha256AssertionRef
+    ]);
+    expect(upsertBlobFromTextMock).toHaveBeenCalledTimes(1);
+    expect(upsertBlobFromTextMock).toHaveBeenCalledWith(
+      blobServiceMock,
+      LOLLIPOP_ASSERTION_STORAGE_CONTAINER_NAME,
+      `${aValidActivatePubKeyPayload.fiscal_code}-${aValidSha256AssertionRef}`,
+      ""
+    );
+    expect(upsertMock).toHaveBeenCalledTimes(2);
+    expect(upsertMock).toHaveBeenNthCalledWith(1, {
+      pubKey: aPendingRetrievedPopDocument.pubKey,
+      // the assertion Ref for masterKey is created by the getAllAssertionRefs method
+      assertionRef: assertionRefsResult.right.master,
+      assertionFileName: `${aFiscalCode}-${aValidSha256AssertionRef}`,
+      status: PubKeyStatusEnum.VALID,
+      assertionType: aValidActivatePubKeyPayload.assertion_type,
+      fiscalCode: aValidActivatePubKeyPayload.fiscal_code,
+      expiredAt: aValidActivatePubKeyPayload.expires_at
+    });
+    expect(upsertMock).toHaveBeenNthCalledWith(2, {
+      pubKey: aPendingRetrievedPopDocument.pubKey,
+      assertionRef: aValidSha256AssertionRef,
+      assertionFileName: `${aFiscalCode}-${aValidSha256AssertionRef}`,
+      status: PubKeyStatusEnum.VALID,
+      assertionType: aValidActivatePubKeyPayload.assertion_type,
+      fiscalCode: aValidActivatePubKeyPayload.fiscal_code,
+      expiredAt: aValidActivatePubKeyPayload.expires_at
+    });
+
+    expect(res.kind).toBe("IResponseSuccessJson");
+    expect(res).toMatchObject({
+      kind: "IResponseSuccessJson",
+      value: retrievedLollipopKeysToApiActivatedPubKey(
+        aValidRetrievedPopDocument
+      )
+    });
+  });
+
+  it("should success given valid informations when used algo == master algo", async () => {
```

Comment:
Personally I'd prefer `equals to` instead of `==` in a title. Just a personal opinion though.

Reply:
done in https://github.com/pagopa/io-functions-lollipop/pull/13/commits/b5d5d231fff8ca9e7201d2acf445b0ddaca4a880